python/lsst/ap/association/diaPipe.py

```diff
@@ -0,0 +1,228 @@
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+"""PipelineTask for associating DiaSources with previous DiaObjects.
+
+Additionally performs forced photometry on the calibrated and difference
+images at the updated locations of DiaObjects.
+
+Currently loads directly from the Apdb rather than pre-loading.
+"""
+
+import os
+
+import lsst.dax.apdb as daxApdb
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as connTypes
```

Comment:
Could you spell out the full `connectionTypes` (as is used in e.g. pipe_tasks) instead of shortening it to `connTypes`?

Reply:
Maybe? I saw a few pipe_tasks that used `cT` instead of `connectionTypes` thought this was a good compromise?